packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs

```diff
@@ -183,9 +187,9 @@ Resources:
   # Marked as depending on UserPoolClientRole for easier to understand CFN sequencing
     Type: 'AWS::IAM::Policy'
     Properties:
-      PolicyName: !Ref userpoolClientLambdaPolicy
+      PolicyName: !If [ShouldNotCreateEnvResources, !Ref userpoolClientLambdaPolicy, !Join ['',[!Ref userpoolClientLambdaPolicy, '-', !Ref env]]]
```

Comment:
You don't need to make the policies unique. Just the role-names need to be unique.

Reply:
changed back